pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```
@@ -328,27 +239,10 @@ func injectSecurityClientLibraryConfig(pod *corev1.Pod) {
 // * "false" - profiling deactivated, not overridable remotely
 // * "auto" - profiling activates per-process heuristically, not overridable remotely
 func injectProfilingClientLibraryConfig(pod *corev1.Pod) {
-	injectEnvVarIfStringConfigKeySet(pod, "admission_controller.auto_instrumentation.profiling.enabled", "DD_PROFILING_ENABLED")
-}
-
-func injectEnvVarIfBoolConfigKeySet(pod *corev1.Pod, configKey string, envVarKey string) {
-	if config.Datadog().IsSet(configKey) {
-		enabledValue := config.Datadog().GetBool(configKey)
-		_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
-			Name:  envVarKey,
-			Value: strconv.FormatBool(enabledValue),
-		})
-	}
-}
-
-func injectEnvVarIfStringConfigKeySet(pod *corev1.Pod, configKey string, envVarKey string) {
-	if config.Datadog().IsSet(configKey) {
-		configValue := config.Datadog().GetString(configKey)
-		_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
-			Name:  envVarKey,
-			Value: configValue,
-		})
-	}
+	_ = configKeyEnvVarMutator{
```

return the error, don't swallow it.